src/components/RegistryViewer.js

```diff
@@ -0,0 +1,60 @@
+import React, { Component } from 'react'
+import ProductionList from "./ProductionList.js"
+import setContractDefaults from "./setContractDefaults.js"
+const loadContract = require("core").loadContract
+
+// Registry: 0xe78a0f7e598cc8b0bb87894b0f60dd2a88d6a8ab
+// Bob: 0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1
+// Alice: 0xffcf8fdee72ac11b5c542428b35eef5769c409f0
+
+
+function applyConstraints(Production, prodAddr, constraints, from)
+{
+	return (Production.at(prodAddr).then((prod) => {
+		return (prod.buyer().then((prodBuyer) => {
+			if ((constraints.buyer && prodBuyer === from) ||
+					(!constraints.buyer && prodBuyer !== from))
```

Comment:
if you want to see all offers, just remove the constraints prop or the buyer element